### DISCUSSION TITLE
Other agents' observation information in the reward

### DISCUSSION THREAD

**Discussion by rsuwa** [praveen-palanisamy/macad-gym#54] [Q&A]
I want to include other agents' observation information in the reward.
Is there a way to do that?

**Comment by praveen-palanisamy** [comment]
Hi @rsuwa ,

If you would like to include more information about the environment/other-agent's observation, you can subclass the base environment and create new environments to suit your needs.
Currently, to compute the reward for an agent, a set of information about the agent including it's interaction (collision info etc) with other agents are available. Although the reward computation only makes use of a subset of this information, the complete set of available information is shown below:
https://github.com/praveen-palanisamy/macad-gym/blob/38884ac4bc7fb2e91a865950cff4eeadeed02f60/src/macad_gym/carla/multi_env.py#L1235-L1260

 If you want to extend the [reward calculation](https://github.com/praveen-palanisamy/macad-gym/blob/master/src/macad_gym/carla/reward.py) using the above available information, you can follow [this Wiki page to create a custom environment](https://github.com/praveen-palanisamy/macad-gym/wiki/creating-new-learning-environments.md)

**Comment by rsuwa** [comment]
@praveen-palanisamy

Thank you for your advice!

I have one more question. For example, if I set the `send_measurements` flag to `True`, I assume I can use other agents' information as observations. Is there a sample code for that somewhere?

https://github.com/praveen-palanisamy/macad-gym/blob/master/src/macad_gym/carla/multi_env.py#L914-L923
